BLSQ/iaso
# Polio 1217 reasons for delay interface

Explain what problem this PR is resolving

Related JIRA tickets :  POLIO-1217

## Self proofreading checklist


## Changes

- Add dependency to django-translated-fields
- Add `ReasonForDelay`model:
- Add `reason_for_delay` ForeignKeyField to `RoundDatesHistoryEntry`
- Deprecate `reason`field.
- Add API endpoint to GET, POST and PATCH
- Replace `reason` with `reason_for_delay` in existing front-end code (`ReasonForDelayModal` mostly)
- Split Polio > Config menu entry in 2 sub menus: country and reasons for delay
- Add Reasons for delay page:
   - Create button: create new reason. `
       - Key` field is mandatory and has regex validation and an info popper to clarify constraints to users
       - English translation is mandatory and will be used as fallback when no translation for French is provided
   - Table view: edit button available. Same as Create, without the info popover. Key field is not editable 

## How to test

- Go to Polio > Configurations > Reasons for delay. Try:
    - Adding
    - Editing
    - Saving without FR translation
- Go to Polio > Campaigns. try to edit a round with the reason just added
- Also check the round dates history table


## Print screen / video

https://github.com/BLSQ/iaso/assets/38907762/ee07c37b-cdd5-4a0b-b407-378daf2ac4f0


## Notes

**IMPORTANT** PowerBI dashboards will need to be updated to use `reason_for_delay` i.o. `reason`. Other than that nothing should change

Reviewers: Please have a careful look at the migration as it was a bit tricky and there's prod data to migrate
